ui/app/services/path-help.js

```diff
@@ -24,37 +29,206 @@ export default Service.extend({
     });
   },
 
+  getNewModel(modelType, owner, backend, apiPath, itemType) {
+    const modelName = `model:${modelType}`;
+    const modelFactory = owner.factoryFor(modelName);
+    let newModel, helpUrl;
+    //if we have a factory, we need to take the existing model into account
+    if (modelFactory) {
+      newModel = modelFactory.class;
+      const modelProto = newModel.proto();
+      if (newModel.merged || modelProto.useOpenAPI !== true) {
+        return resolve();
+      }
+      helpUrl = modelProto.getHelpUrl(backend);
+      return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName, owner);
+    } else {
+      newModel = DS.Model.extend({});
+      //use paths to dynamically create our openapi help url
+      //if we have a brand new model
+      return this.getPaths(apiPath, backend, itemType).then(paths => {
+        const adapterFactory = owner.factoryFor(`adapter:${modelType}`);
+        //if we have an adapter already use that, otherwise create one
+        if (!adapterFactory) {
+          const adapter = this.getNewAdapter(backend, paths, itemType);
+          owner.register(`adapter:${modelType}`, adapter);
+        }
+        //if we have an item we want the create info for that itemType
+        if (itemType) {
+          let { tag, path } = paths.create[0];
+          path = path.slice(0, path.indexOf('{') - 1) + '/example';
+          helpUrl = `/v1/${tag}/${backend}${path}?help=true`;
+        } else {
+          //we need the mount config
+          let { tag, path } = paths.configPath[0];
+          helpUrl = `/v1/${tag}/${backend}${path}?help=true`;
+        }
+        return this.registerNewModelWithProps(helpUrl, backend, newModel, modelName, owner);
+      });
+    }
+  },
+
+  getPaths(apiPath, backend, itemType) {
+    return this.ajax(`/v1/${apiPath}?help=1`, backend).then(help => {
+      const pathInfo = help.openapi.paths;
+      let paths = Object.keys(pathInfo);
+      //remove deprecated endpoints
+      paths = paths.filter(path => pathInfo[path]['x-vault-sudo'] !== true); //get rid of deprecated paths
```

Comment:
does 'x-vault-sudo' mean the endpoints are deprecated?

Reply:
It doesn't. Instead of excluding endpoints from navigation this way, we're going to explicitly include them with displayAttrs